test/go/table_schema_cache_test.go

```diff
@@ -65,30 +76,159 @@ func (this *TableSchemaCacheTestSuite) TestLoadTablesWithoutFiltering() {
 }
 
 func (this *TableSchemaCacheTestSuite) TestLoadTablesRejectTablesWithoutNumericPK() {
-	query := fmt.Sprintf("CREATE TABLE %s.%s (id varchar(20) not null, data TEXT, primary key(id))", testhelpers.TestSchemaName, "test_table_4")
+	table := "test_table_4"
+	paginationColumn := "id"
+	query := fmt.Sprintf("CREATE TABLE %s.%s (%s varchar(20) not null, data TEXT, primary key(%s))", testhelpers.TestSchemaName, table, paginationColumn, paginationColumn)
+	_, err := this.Ferry.SourceDB.Exec(query)
+	this.Require().Nil(err)
+
+	_, err = ghostferry.LoadTables(this.Ferry.SourceDB, this.tableFilter, nil, nil, nil)
+
+	this.Require().NotNil(err)
+	this.Require().EqualError(err, ghostferry.NonNumericPaginationKeyError(testhelpers.TestSchemaName, table, paginationColumn).Error())
+	this.Require().Contains(err.Error(), table)
+}
+
+func (this *TableSchemaCacheTestSuite) TestLoadTablesCascadingPaginationColumnConfigRightScenario1() {
+	table := "pagination_by_column_config_right_scenario_1"
+	paginationColumn := "identity"
+	cascadingPaginationColumnConfig := &ghostferry.CascadingPaginationColumnConfig{
+		PerTable: map[string]map[string]string{
+			testhelpers.TestSchemaName: map[string]string{
+				table: paginationColumn,
+			},
+		},
+	}
+
+	query := fmt.Sprintf("CREATE TABLE %s.%s (%s bigint(20) not null, data TEXT)", testhelpers.TestSchemaName, table, paginationColumn)
+	_, err := this.Ferry.SourceDB.Exec(query)
+	this.Require().Nil(err)
+
+	_, err = ghostferry.LoadTables(this.Ferry.SourceDB, this.tableFilter, nil, nil, cascadingPaginationColumnConfig)
```

Comment:
Maybe we can quickly assert that the table `pagination_by_column_config_right_scenario_1` does indeed have the right PaginationColumn set? Is this necessary?